plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryInstanceCleaner.java

```diff
@@ -0,0 +1,146 @@
+    {
+        LOG.info("Identifying tables to drop...");
+        // Drop all tables created before yesterday (in UTC)
+        TableResult result = bigQuerySqlExecutor.executeQuery(format("" +
+                        "SELECT table_name, table_type " +
+                        "FROM %s.INFORMATION_SCHEMA.TABLES " +
+                        "WHERE date_diff(current_date(), date(creation_time), DAY) > 1 " +
+                        "AND table_name NOT IN (%s)" +
+                        "AND table_type IN (%s)",
+                quoted(schemaName),
+                tablesToKeep.stream().collect(joining("','", "'", "'")),
+                tableTypesToDrop.stream().collect(joining("','", "'", "'"))));
+        List<Entry<String, String>> objectsToDrop = Streams.stream(result.getValues())
+                .map(fieldValues -> new SimpleImmutableEntry<>(fieldValues.get("table_name").getStringValue(), fieldValues.get("table_type").getStringValue()))
+                .collect(toImmutableList());
+
+        if (objectsToDrop.isEmpty()) {
+            LOG.info("Did not find any objects to drop.");
+            return;
+        }
+
+        LOG.info("Dropping %s objects.", objectsToDrop.size());
+        LOG.info("Dropping: %s", objectsToDrop.stream().map(Entry::getKey).collect(joining(", ")));
+        objectsToDrop.forEach(entry -> {
+            String dropStatement = getDropStatement(schemaName, entry.getKey(), entry.getValue());
+            LOG.info("Executing: %s", dropStatement);
+            bigQuerySqlExecutor.execute(dropStatement);
+        });
+    }
+
+    private void logObjectsCount(String schemaName)
+    {
+        TableResult result = bigQuerySqlExecutor.executeQuery(format("" +
+                        "SELECT table_type, count(*) AS c " +
+                        "FROM %s.INFORMATION_SCHEMA.TABLES " +
+                        "GROUP BY table_type",
+                quoted(schemaName)));
+        result.getValues().forEach(fieldValues -> LOG.info("Schema '%s' contains '%s' objects of type '%s'",
+                schemaName,
+                fieldValues.get("c").getLongValue(),
+                fieldValues.get("table_type").getStringValue()));
+    }
+
+    private static String getDropStatement(String schemaName, String objectName, String objectType)
+    {
+        switch (objectType) {
+            case "BASE TABLE":
+                return format("DROP TABLE %s.%s", quoted(schemaName), quoted(objectName));
+            case "VIEW":
+                return format("DROP VIEW %s.%s", quoted(schemaName), quoted(objectName));
+            case "MATERIALIZED VIEW":
+                return format("DROP MATERIALIZED VIEW %s.%s", quoted(schemaName), quoted(objectName));
+            default:
+                throw new IllegalArgumentException("Unexpected object type " + objectType);
+        }
+    }
+
+    private static String quoted(String identifier)
+    {
+        return "`" + identifier.replace("`", "\\`") + "`";
```

Comment:
if `\` is the escape character, then `\` needs to be replaced as well

however, can we use "SQL mode" in BigQuery, ie use `"`-quoting?

Reply:
No, BigQuery only uses ` for quoted identifiers. Changed to also escape backslashes.